### ISSUE TITLE
TLS Connection improvements

### ISSUE THREAD

**Issue by ehsandeep** [projectdiscovery/tlsx#51]
Example host: `209.17.116.160`

TLSX Run:

```console
echo 209.17.116.160 | tlsx -silent -v

[INF] Processing input 209.17.116.160:443
[WRN] Could not connect input 209.17.116.160:443: could not connect to host: could not do handshake: remote error: tls: internal error
```

As we see above, `209.17.116.160` failed to connect but also available as part of shodan scan shared below:

Shodan: https://www.shodan.io/host/209.17.116.160#443

**Comment by tarunKoyalwar** [comment]
also happens with `openssl` 
```sh
openssl s_client -connect  209.17.116.160:443
CONNECTED(00000003)
803A07FB01000000:error:0A000438:SSL routines:ssl3_read_bytes:tlsv1 alert internal error:ssl/record/rec_layer_s3.c:1600:SSL alert number 80
---
no peer certificate available
---
No client certificate CA names sent
---
SSL handshake has read 7 bytes and written 297 bytes
Verification: OK
---
New, (NONE), Cipher is (NONE)
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
Early data was not sent
Verify return code: 0 (ok)
---

```

**Comment by tarunKoyalwar** [comment]
also happens with all ips of `projectdiscovery.io` but error is different

**Comment by tarunKoyalwar** [comment]
According to https://github.com/openssl/openssl/issues/7147#issuecomment-419621673 this happens due to missing `Hostname` when server has more than 1 hostname . if servername/SNI is specified it connects successfully

```sh
 openssl s_client -connect  209.17.116.160:443 -servername lssjax.org
CONNECTED(00000003)
depth=2 C = US, ST = New Jersey, L = Jersey City, O = The USERTRUST Network, CN = USERTrust RSA Certification Authority
verify error:num=19:self-signed certificate in certificate chain
verify return:1
depth=2 C = US, ST = New Jersey, L = Jersey City, O = The USERTRUST Network, CN = USERTrust RSA Certification Authority
verify return:1
depth=1 C = US, ST = VA, L = Herndon, O = Network Solutions L.L.C., CN = Network Solutions DV Server CA 2
verify return:1
depth=0 CN = lssjax.org
verify return:1
---
Certificate chain
 0 s:CN = lssjax.org
   i:C = US, ST = VA, L = Herndon, O = Network Solutions L.L.C., CN = Network Solutions DV Server CA 2
   a:PKEY: rsaEncryption, 2048 (bit); sigalg: RSA-SHA256
...
```

**Comment by Mzack9999** [comment]
Potentially this is indirectly fixed by https://github.com/projectdiscovery/tlsx/issues/142

**Comment by hktalent** [comment]
You can use my website for testing
```
$ping 51pwn.com
64 bytes from 172.67.210.144: icmp_seq=0 ttl=47 time=236.726 ms

$openssl s_client -connect 172.67.210.144:443|openssl x509 -pubkey -noout 
140704302261056:error:14094410:SSL routines:ssl3_read_bytes:sslv3 alert handshake failure:ssl/record/rec_layer_s3.c:1563:SSL alert number 40
unable to load certificate
140704302261056:error:0909006C:PEM routines:get_name:no start line:crypto/pem/pem_lib.c:745:Expecting: TRUSTED CERTIFICATE

```

**Comment by ehsandeep** [comment]
@hktalent it works with tlsx as well

```console
echo 51pwn.com | tlsx
  

  _____ _    _____  __
 |_   _| |  / __\ \/ /
   | | | |__\__ \>  < 
   |_| |____|___/_/\_\	v1.1.1

		projectdiscovery.io

[INF] Current tlsx version v1.1.1 (latest)
51pwn.com:443
[INF] Connections made using crypto/tls: 1, zcrypto/tls: 0, openssl: 0
```

**Comment by hktalent** [comment]
@ehsandeep In most cases, using 
echo 11.11.11.0/16 | tlsx

Performing a large-scale scan in the form of SSL 1.3 can result in the loss of a lot of useful information, especially in the case of 51pwn.com, which uses SSL 1.3
